src/coreclr/jit/valuenum.cpp

```diff
@@ -11835,7 +11835,36 @@ void Compiler::fgValueNumberHWIntrinsic(GenTreeHWIntrinsic* tree)
         }
     }
 
-    tree->gtVNPair = vnStore->VNPWithExc(normalPair, excSetPair);
+    // Some intrinsics should always be unique
+    bool makeUnique = false;
+
+#if defined(TARGET_XARCH)
+    switch (intrinsicId)
+    {
+        case NI_AVX512F_ConvertMaskToVector:
+        {
+            // We want to ensure that we get a TYP_MASK local to
+            // ensure the relevant optimizations can kick in
```

Comment:
This ensures we don't end up with a local hiding the `ConvertMaskToVector` and results in overall better codegen since most cases allow it to be elided.

Reply:
And this is not needed for `ConvertVectorToMask`?

Reply:
Right. We just care about seeing the `TYP_MASK` node since it needs special handling. `TYP_SIMD` nodes are already well handled throughout.